File: gameserver/src/main/java/l2s/gameserver/network/authcomm/gs2as/BonusRequest.java
package l2s.gameserver.network.authcomm.gs2as;

import io.netty.buffer.ByteBuf;
import l2s.gameserver.network.authcomm.SendablePacket;

public class BonusRequest extends SendablePacket {
    private final String account;
    private final int bonus;
    private final int bonusExpire;

    public BonusRequest(String account, int bonus, int bonusExpire) {
        this.account = account;
        this.bonus = bonus;
        this.bonusExpire = bonusExpire;
    }

    @Override
    protected ByteBuf writeImpl(ByteBuf byteBuf) {
        writeString(byteBuf, account);
        byteBuf.writeInt(bonus);
        byteBuf.writeInt(bonusExpire);
        return byteBuf;
    }

    @Override
    protected byte getOpCode() {
        return 16;
    }
}

File: gameserver/src/main/java/l2s/gameserver/network/authcomm/gs2as/BonusRequest.kt
package l2s.gameserver.network.authcomm.gs2as

import io.netty.buffer.ByteBuf
import l2s.gameserver.network.authcomm.SendablePacket

class BonusRequest(private val account: String, private val bonus: Int, private val bonusExpire: Int) :
    SendablePacket() {

    override val opCode: Byte = 0x10

    override fun writeImpl(byteBuf: ByteBuf): ByteBuf {
        writeString(byteBuf, account)
        byteBuf.writeInt(bonus)
        byteBuf.writeInt(bonusExpire)

        return byteBuf
    }

}